tests/test_integration.py

```diff
@@ -92,3 +88,59 @@ def test_agg(main_summary, search_clients_daily, events, raw_pings, spark):
     result = df_to_json(agg)
     expected_result = load_json("expected_results.json")
     assert result == expected_result
+
+
+@pytest.fixture()
+def mock_main_dependencies(
+    main_summary, search_clients_daily, events, raw_pings, spark, monkeypatch
+):
+    def mock_get_spark(*args, **kwargs):
+        return spark
+
+    def mock_load_data_s3(*args, **kwargs):
+        prefix = kwargs["input_prefix"]
+        if prefix == "main_summary":
+            df = main_summary.withColumn("normalized_channel", F.lit("release"))
+        elif prefix == "search_clients_daily":
+            df = search_clients_daily
+        elif prefix == "events":
+            df = events
+        else:
+            raise NotImplementedError
+        return df.withColumn("sample_id", F.lit("0"))
+
+    def mock_load_raw_pings(*args, **kwargs):
+        return raw_pings
+
+    monkeypatch.setattr(addons_daily.addons_report, "get_spark", mock_get_spark)
+    monkeypatch.setattr(addons_daily.addons_report, "load_data_s3", mock_load_data_s3)
+    monkeypatch.setattr(
+        addons_daily.addons_report, "load_raw_pings", mock_load_raw_pings
+    )
+
+
+def test_main_end_to_end(mock_main_dependencies, tmpdir):
+    runner = CliRunner()
+    result = runner.invoke(
+        main, ["--date", BASE_DATE, "--output", "file://" + str(tmpdir)]
+    )
+    assert result.exit_code == 0
+    submission_path = tmpdir / "submission_date_s3={}".format(BASE_DATE)
+    assert os.path.exists(submission_path)
+    assert len([p for p in os.listdir(submission_path) if p.endswith(".parquet")]) > 0
+
+
+def test_main_invalid_prefix(mock_main_dependencies, tmpdir):
+    runner = CliRunner()
+
+    # missing "file://"
+    result = runner.invoke(main, ["--date", BASE_DATE, "--output", str(tmpdir)])
+    assert result.exit_code == 1
+    assert isinstance(result.exception, ValueError)
+
+    # not a supported path
```

Comment:
This can be removed when writing to google storage is standard. I'm not sure if Databricks supports this pathway.